www_src/components/modal/modal.jsx

```diff
@@ -0,0 +1,67 @@
+var React = require('react');
+var dispatcher = require('../../lib/dispatcher');
+
+var Modal = React.createClass({
+  getInitialState: function () {
+    return {
+      isActive: false,
+      header: '',
+      body: '',
+      attribution: undefined,
+      icon: '',
+      buttonText: 'OK, got it!',
+      callback: null
+    };
+  },
+  show: function () {
+    this.setState({
+      isActive: true
+    });
+  },
+  hide: function () {
+    this.setState({
+      isActive: false
+    });
+  },
+  onConfirmClick: function () {
+    this.hide();
+
+    if (this.state.callback) {
+      this.state.callback.call();
+    }
+  },
+  componentDidMount: function () {
+    dispatcher.on('modal:show', (event) => {
+      this.setState(event.config);
+      this.show();
+    });
+
+    dispatcher.on('modal:hide', (event) => {
+      this.hide();
```

Comment:
Otherwise subsequent modals will pick up the last model's config instead of defaults -- either that, or have assign(this.getInitialState, event.config) in the `modal:show` event